code/_onclick/hud/parallax.dm

```diff
@@ -194,18 +196,27 @@
 		var/obj/screen/parallax_layer/L = thing
 		if (L.view_sized != C.view)
 			L.update_o(C.view)
-		var/change_x = offset_x * L.speed
-		L.offset_x -= change_x
-		var/change_y = offset_y * L.speed
-		L.offset_y -= change_y
-		if(L.offset_x > 240)
-			L.offset_x -= 480
-		if(L.offset_x < -240)
-			L.offset_x += 480
-		if(L.offset_y > 240)
-			L.offset_y -= 480
-		if(L.offset_y < -240)
-			L.offset_y += 480
+
+		var/change_x
+		var/change_y
+
+		if(L.absolute)
+			L.offset_x = -(posobj.x - SSparallax.planet_x_offset) * L.speed
+			L.offset_y = -(posobj.y - SSparallax.planet_y_offset) * L.speed
```

Comment:
Make a new /obj/screen/parallax_layer proc that does the normal behavior then override it in the planet version.

Reply:
proc overhead

Reply:
If we're going to be having parallax layers act differently, putting the different behaviors in the generic loop seems a bad move

Reply:
This is something that can be applied to other parallax layers, and the behaviour would be useful. The argument for including it as a var instead makes more sense IMO and parallax loops need to be fairly performing

Reply:
That's fair enough, we can switch it over easily enough I guess if we end up needing to do this.